core/src/main/scala/com/spotify/featran/transformers/NGramsTransformer.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.transformers
+
+import com.spotify.featran.FeatureBuilder
+
+import scala.collection.SortedMap
+
+/**
+ * Transform a collection of sentences, where each row is a `Seq[String]` of the words / tokens,
+ * into a collection containing all the n-grams that can be constructed from each row. The feature
+ * representation is an m-hot encoding (see [[NHotEncoder]]) constructed from an expanded vocabulary
+ * of all of the generated n-grams.
+ *
+ * N-grams are generated based on a specified range of `low` to `high` (inclusive) and are joined by
+ * the given `separator` (the default is ""). For example, an [[NGramsTransformer]] with
+ * `separator = ""`, `low = 2`, and `high = 3`, the row `["a", "b", "c", "d", "e"]` would produce
+ * the ngrams `["ab", "bc", "cd", "de", "abc", "bcd", "cde"]`.
+ *
+ * As with [[NHotEncoder]], missing values are transformed to [0.0, 0.0, ...].
+ */
+object NGramsTransformer {
+  type TfType = Transformer[Seq[String], Set[String], SortedMap[String, Int]]
+
+  /**
+   * Create a new [[NGramsTransformer]] instance.
+   * @param separator a string separator used to join individual tokens
+   * @param low the smallest size of the generated *-grams
+   * @param high the largest size of the generated *-grams
+   */
+  def apply(name: String, separator: String, low: Int, high: Int): TfType =
```

Comment:
`def apply(name: String, low: Int = 1, high: Int = -1, sep: String = " ")`? This way 1 `apply` handles all. Just need to make it clear that by default it generates n-grams all the way to `n = xs.length` and may be expensive.